spec/pdfs/siren_infos_pdf_spec.rb

```diff
@@ -0,0 +1,101 @@
+    # Identité PM
+    it { is_expected.to include('SIREN: 111 111 111') }
+    it { is_expected.to include('Date d\'immatriculation: 1968-05-02') }
+    it { is_expected.to include('Dénomination: Willy Wonka Candies Factory') }
+    it { is_expected.to include('Forme juridique: Société de bonbons à responsabilité limitée') }
+    it { is_expected.to include('Capital: 1000.0 Euros') }
+    it { is_expected.to include('Adresse: Rue des cocotiers 97114 Trois-Rivières') }
+    it { is_expected.to include('Durée: 22/10/2117') }
+    it { is_expected.to include('Date de clôture: 31 Décembre') }
+    # Représentant
+    it { is_expected.to include('Qualité: Président') }
+    it { is_expected.to include('Dénomination: Grosse Entreprise de Télécom') }
+    it { is_expected.to include('SIREN: 333 444 555') }
+    it { is_expected.to include('Forme juridique: Société par actions simplifiée') }
+    it { is_expected.to include('Adresse:  rue des Peupliers Zone Industrielle Sud 34000 Montpellier') }
+  end
+
+  describe 'Entreprise complexe' do
+    let(:dossier) { create :entreprise_complexe }
+
+    # Greffe
+    it { is_expected.to include('Greffe entreprise complexe (code greffe: 1234)') }
+    it { is_expected.to include('Numéro de gestion: 1968A00666') }
+    # Identité PM
+    it { is_expected.to include('SIREN: 222 222 222') }
+    it { is_expected.to include('Date d\'immatriculation: 1968-05-02') }
+    it { is_expected.to include('Dénomination: Willy Wonka Candies Factory') }
+    it { is_expected.to include('Forme juridique: Société de bonbons à responsabilité limitée') }
+    it { is_expected.to include('Capital: 1000.0 Euros') }
+    it { is_expected.to include('Adresse: Rue des cocotiers 97114 Trois-Rivières') }
+    it { is_expected.to include('Durée: 22/10/2117') }
+    it { is_expected.to include('Date de clôture: 31 Décembre') }
+    # Représentant*S*
+    ## Président
+    it { is_expected.to include('Qualité: Président') }
+    it { is_expected.to include('Dénomination: Grosse Entreprise de Télécom') }
+    it { is_expected.to include('SIREN: 333 444 555') }
+    it { is_expected.to include('Forme juridique: Société par actions simplifiée') }
+    it { is_expected.to include('Adresse:  rue des Peupliers Zone Industrielle Sud 34000 Montpellier') }
+    ## Autres
+    it 'has more représentants' do
+      nb_rep = dossier.representants.size
+      _, reps = subject.split('Représentants')
+
+      expect(reps.size).to eq(nb_rep*5)
+    end
+  end
+
+  describe 'all fields are nil' do
+    let(:dossier) { DossierEntreprise.new.tap(&:save) }
+
+    before { Etablissement.new(dossier_entreprise: dossier).tap(&:save) }
+
+    context 'with PP' do
+      before { PersonnePhysique.new(dossier_entreprise: dossier).tap(&:save) }
+
+      its(:size) { is_expected.to eq 11 }
+    end
+
+    context 'with PM' do
+      before { PersonneMorale.new(dossier_entreprise: dossier).tap(&:save) }
+
+      its(:size) { is_expected.to eq 14 }
+    end
+  end
```

Comment:
Pas très clair, je ne comprends pas ce qui est testé ici, ni comment

Reply:
Je cherche à tester que lorsque certaines (ici tous) champs sont nuls qu'on génère quand même le PDF et que rien ne plante.

Je teste donc qu'on a bien x lignes de chaînes de caractères dans le PDF. Je ne voulais pas alourdir les specs en remettant 14 fois un truc du genre : 
`it { is_expected.to include('SIREN: ') }`

Mais je peux le faire si c'est plus clair

Reply:
Euh je suis toujours pour une suppression du test...
On ne va pas tester qu'on est capable de générer un PDF avec toutes les valeurs à nil alors que nos spécifications consistent justement à ne pas générer de PDF dans ce cas là

Reply:
solution opté : 

Le PDF peut renvoyer une erreur si le dossier demandé ne permet pas de générer un document valide.

Donc le controller dois filtrer en amont les dossiers à rejeter <= TODO